File: is-palindrome/Palindrome.java
public class Palindrome {
    public static boolean isPalindrome1(String str) {
        int left = 0;
        int right = str.length() - 1;
        while(left < right) {
            if(Character.toLowerCase(str.charAt(left)) != Character.toLowerCase(str.charAt(right))) {
                return false;
            }
            left++; right--;
        }
        return true;
    }

    public static boolean isPalindrome2(String str) {
        int left = 0;
        int right = str.length() - 1;
        while(left < right) {
            while(left < right && !Character.isLetterOrDigit(str.charAt(left))) { left++; }
            while(left < right && !Character.isLetterOrDigit(str.charAt(right))) { right--; }
            if(Character.toLowerCase(str.charAt(left)) != Character.toLowerCase(str.charAt(right))) return false;
            left++; right--;
        }
        return true;
    }
}

File: is-palindrome/Palindrome.js
function palindrome1(str) {
    let left = 0;
    let right = str.length - 1;
    while(left < right) {
        if(str.charAt(left).toLowerCase() !== str.charAt(right).toLowerCase()){
            return false;
        }
    }
    return true;
}

function palindrome(str) {
    let left = 0;
    let right = str.length - 1;
    while(left < right) {
        while(left < right && !isLetterOrDigit(str.charAt(left))) { left++; }
        while(left < right && !isLetterOrDigit(str.charAt(right))) { right--; }
        if(str.charAt(left).toLowerCase() !== str.charAt(right).toLowerCase()) return false;
        left++; right--;
    }
    return true;
}

var isLetterOrDigit = (character) => {
    if(typeof character !== "string") return false;
    return character.toLowerCase() !== character.toUpperCase();
}

exports.palindrome = palindrome;